pkg/clients/atlan/teams.go

```diff
@@ -0,0 +1,125 @@
+package atlan
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/redhat-data-and-ai/usernaut/pkg/common/structs"
+	"github.com/redhat-data-and-ai/usernaut/pkg/logger"
+	"github.com/sirupsen/logrus"
+)
+
+type AtlanGroup struct {
+	ID   string `json:"id"`
+	Name string `json:"name"`
+}
+
+type AtlanGroupsResponse struct {
+	TotalRecord  int          `json:"totalRecord"`
+	FilterRecord int          `json:"filterRecord"`
+	Records      []AtlanGroup `json:"records"`
+}
+
+func (ac *AtlanClient) FetchAllTeams(ctx context.Context) (map[string]structs.Team, error) {
+	log := logger.Logger(ctx).WithField("service", "atlan")
+	log.Info("fetching all teams from Atlan")
+
+	url := fmt.Sprintf("%s/api/service/v2/groups?columns=name", ac.url)
+	response, statusCode, err := ac.sendRequest(ctx, url, http.MethodGet, nil, nil, "FetchAllTeams")
```

Comment:
any reason we are using http client for atlan? can we use their sdk instead? Just went through their release notes: https://github.com/atlanhq/atlan-go/releases/tag/v0.0.13, this has the capability to manage groups. Please do let me know your thoughts on this.